x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality/dataset_quality.tsx

```diff
@@ -25,45 +31,36 @@ export interface CreateDatasetQualityArgs {
   telemetryClient: ITelemetryClient;
 }
 
-export const createDatasetQuality = ({
+export const DatasetQuality = ({
+  controller,
   core,
   plugins,
   telemetryClient,
-}: CreateDatasetQualityArgs) => {
-  return ({ controller }: DatasetQualityProps) => {
-    const SummaryPanelProvider = dynamic(() => import('../../hooks/use_summary_panel'));
-    const KibanaContextProviderForPlugin = useKibanaContextForPluginProvider(core, plugins);
+}: DatasetQualityProps & CreateDatasetQualityArgs) => {
+  const KibanaContextProviderForPlugin = useKibanaContextForPluginProvider(core, plugins);
 
-    const datasetQualityProviderValue: DatasetQualityContextValue = useMemo(
-      () => ({
-        service: controller.service,
-        telemetryClient,
-      }),
-      [controller.service]
-    );
+  const datasetQualityProviderValue: DatasetQualityContextValue = useMemo(
+    () => ({
+      service: controller.service,
+      telemetryClient,
+    }),
+    [controller.service, telemetryClient]
+  );
 
-    return (
-      <PerformanceContextProvider>
-        <DatasetQualityContext.Provider value={datasetQualityProviderValue}>
-          <SummaryPanelProvider>
-            <KibanaContextProviderForPlugin>
-              <DatasetQuality />
-            </KibanaContextProviderForPlugin>
-          </SummaryPanelProvider>
-        </DatasetQualityContext.Provider>
-      </PerformanceContextProvider>
-    );
-  };
+  return (
+    <PerformanceContextProvider>
+      <DatasetQualityContext.Provider value={datasetQualityProviderValue}>
+        <SummaryPanelProvider>
+          <KibanaContextProviderForPlugin>
+            <DatasetQualityContent />
+          </KibanaContextProviderForPlugin>
+        </SummaryPanelProvider>
+      </DatasetQualityContext.Provider>
+    </PerformanceContextProvider>
+  );
 };
 
-const Header = dynamic(() => import('./header'));
-const Warnings = dynamic(() => import('./warnings/warnings'));
-const EmptyStateWrapper = dynamic(() => import('./empty_state/empty_state'));
-const Table = dynamic(() => import('./table/table'));
-const Filters = dynamic(() => import('./filters/filters'));
-const SummaryPanel = dynamic(() => import('./summary_panel/summary_panel'));
```

Comment:
> [!NOTE]
> All of these chunks were loaded in parallel when loading the page, causing an excessive fragmentation for a UI that can loaded at once.
>
> The lazy loading now happens for the main chunk on the index file, where the getter dynamically import the whole UI island.

Reply:
to better understand what happened here:
Should we use this approach then only for elements in the UI that are showed after some interaction? for example a flyout?

Reply:
Sort of, the best way to think about this is to split only when we might not need something.
For instance, in the above case we were splitting into 6 chunks sections of the same page that are always immediately loaded when the page is visited. In this case we could consider it as a single "UI island" that can be dynamically loaded at once, reducing the back and forth with the server.

The flyout example you are doing is a good one, as the user might never open the flyout during a session.